src/app/sidebar-tree/sidebar-tree.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {MatTreeNestedDataSource} from '@angular/material/tree';
+import {NestedTreeControl} from '@angular/cdk/tree';
+import {Category} from '../category';
+import {of} from 'rxjs';
+import {ActivatedRoute} from '@angular/router';
+
+@Component({
+  selector: 'app-sidebar-tree',
+  templateUrl: './sidebar-tree.component.html',
+  styleUrls: ['./sidebar-tree.component.css']
+})
+export class SidebarTreeComponent implements OnInit, OnDestroy {
+
+  nestedTreeControl: NestedTreeControl<Category>;
+  nestedDataSource: MatTreeNestedDataSource<Category>;
+  categories: [Category];
```

Comment:
Oh, it's unusual way to set array of Category type :) 
```typescript
const categories: Category[]
```
https://www.typescriptlang.org/docs/handbook/basic-types.html#array

You can also add seperate type for array of Category:
```typescript
export type Categories = Category[];
```